package.json

```diff
@@ -4,20 +4,30 @@
   "author": {
     "name": "0xflotus"
   },
-  "main": "dist/index.js",
+  "main": "app/index.js",
+  "types": "app/index.d.ts",
   "license": "MIT",
   "scripts": {
-    "build": "tsc"
+    "prepare": "npm run build",
+    "build": "npm run build:types && npm run build:code",
+    "build:code": "tsc",
+    "build:types": "tsc -d --declarationDir app --declarationMap --emitDeclarationOnly",
```

Comment:
These new scripts allow both Javascript and Typescript users to get full type coverage. When using Typescript the IDE should reference the `.ts` file where as with Javascript it'll reference the generated `.d.ts` file.

Reply:
Thanks. Can you please resolve the conflicts, so I can merge the changes? :)

Reply:
Sure, I'll get to it when I have a spare minute.